framework/platform/bpl/uci/db/bpl_db_uci.cpp

```diff
@@ -144,6 +144,69 @@ bool uci_set_section(const std::string &package_name, const std::string &section
     LOG(TRACE) << "uci_set_section() " << package_name << ".(" << section_type << ")"
                << section_name;
 
+    auto ctx = alloc_context();
+    if (!ctx) {
+        return false;
+    }
+
+    for (auto &option : options) {
+        char opt_path[MAX_UCI_BUF_LEN] = {0};
+        if (snprintf(opt_path, MAX_UCI_BUF_LEN, option_path, package_name.c_str(),
+                     section_name.c_str(), option.first.c_str()) <= 0) {
+            LOG(ERROR) << "Failed to compose path";
+            return false;
+        }
+
+        LOG(TRACE) << "Trying to lookup: " << opt_path;
+
+        struct uci_ptr opt_ptr;
+        // Get pointer to value
+        if (uci_lookup_ptr(ctx.get(), &opt_ptr, opt_path, true) != UCI_OK) {
+            LOG(ERROR) << "UCI failed to lookup ptr for path: " << opt_path << std::endl
+                       << uci_get_error(ctx.get());
+            return false;
+        }
+
+        //Set the value to the pointer
+        LOG(TRACE) << "Setting " << option.first << "=" << option.second;
+        opt_ptr.value = strdup(option.second.c_str());
+
+        /*
+            If the value is set to empty the option will be removed
+            If the option does not exist it will be created
+            If the option does exist the value will be overridden
+        */
+        if (uci_set(ctx.get(), &opt_ptr) != UCI_OK) {
+            LOG(ERROR) << "Failed to set option " << option.first;
+            return false;
+        } else {
```

Comment:
no need for else. the if returns from the function anyway

Reply:
Removed the else